gadopt/approximations.py

```diff
@@ -466,15 +531,204 @@ def prefactor_prestress(self, dt):
+    Lau, H. C. (2018). Quantifying the sensitivity of post-glacial sea level change
+    to laterally varying viscosity. Geophysical journal international, 214(2), 1324-1363.
+
+    Automated forward and adjoint modelling of viscoelastic deformation of the solid
+    Earth.  Scott, W.; Hoggard, M.; Duvernay, T.; Ghelichkhan, S.; Gibson, A.;
+    Roberts, D.; Kramer, S. C.; and Davies, D. R. EGUsphere, 2025: 1–43. 2025.
+
+
+    Arguments:
+      bulk_modulus:  bulk modulus
+      density:       background density
+      shear_modulus: shear modulus
+      viscosity:     viscosity
+      bulk_shear_ratio: Ratio of bulk to shear modulus
+      compressible_buoyancy: Include compressible buoyancy effects
+      compressible_adv_hyd_pre: Include compressible hydrostatic prestress advection
+      g:             gravitational acceleration
+      B_mu:          Nondimensional number describing ratio of buoyancy to elastic
+                     shear strength used for nondimensionalisation.
+                     $ B_{\\mu} = \frac{\bar{\rho} \bar{g} L}{\bar{\\mu}}$,
+                     where $\bar{\rho}$ is a characteristic density scale (kg / m^3),
+                     $\bar{g}$ is a characteristic gravity scale (m / s^2),
+                     $L$ is a characteristic length scale, often Mantle depth (m),
+                     $\\mu$ is a characteristic shear modulus (Pa).
+
+    """
+
+    compressible = True
+
+    def __init__(
+        self,
+        bulk_modulus: Function | Number,
+        density: Function | Number,
+        shear_modulus: Function | Number | list,
+        viscosity: Function | Number | list,
+        *,
+        bulk_shear_ratio: Function | Number = 1,
+        compressible_buoyancy: bool = True,
+        compressible_adv_hyd_pre: bool = True,
+        **kwargs,
+    ):
+        self.bulk_modulus = ensure_constant(bulk_modulus)
+        self.bulk_shear_ratio = ensure_constant(bulk_shear_ratio)
+        self.compressible_buoyancy = compressible_buoyancy
+        self.compressible_adv_hyd_pre = compressible_adv_hyd_pre
+
+        # Convert viscosity and shear modulus to lists in the case
+        # for Maxwell Rheology where there is only one internal variable
+        # and hence only one pair of viscosity and shear modulus fields
+        if not isinstance(shear_modulus, list):
+            shear_modulus = [shear_modulus]
+        if not isinstance(viscosity, list):
+            viscosity = [viscosity]
+
+        if len(viscosity) != len(shear_modulus):
+            raise ValueError("Length of viscosity and shear modulus lists must be consistent")
+
+        super().__init__(density, shear_modulus, viscosity, **kwargs)
+        self.maxwell_times = [ensure_constant(visc / mu) for visc, mu in zip(self.viscosity, self.shear_modulus)]
+        self.mu0 = ensure_constant(sum(self.shear_modulus))
+
+    def div_u(self, u: Function) -> ufl.core.expr.Expr:
+        dim = len(u)
+        return div(u) * Identity(dim)
```

Comment:
I am not convinced this method is necessary. Rather, you could define `self.identity` and use it wherever necessary.

Reply:
this was removed by Dale